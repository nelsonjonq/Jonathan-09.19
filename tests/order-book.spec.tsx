import "jest-enzyme";

import React from "react";

import enzyme from "enzyme";

import Adapter from "enzyme-adapter-react-16";
import ErrorMessage from "../src/components/error-message";
import Header from "../src/components/header";
import OrderAskChart from "../src/components/charts/ask-chart";
import OrderBidChart from "../src/components/charts/bid-chart";
import OrderRow from "../src/components/charts/order-row";
import ToggleProductButton from "../src/components/buttons/toggle-product-button";
import SpreadRow from "../src/components/spread";

import { Pressable } from "react-native";

enzyme.configure({ adapter: new Adapter() });

const mockedMaxTotalSum = 10000;
const getMockedData = () => {
  return [
    [45000, 100],
    [44950, 200],
    [44900, 650],
  ];
};

describe("OrderAskChart", () => {
  it("should render correctly", () => {
    const component = enzyme.shallow(
      <OrderAskChart data={getMockedData()} maxTotalSum={mockedMaxTotalSum} />,
    );

    expect(component).toMatchSnapshot();
  });
});

describe("OrderBidChart", () => {
  it("should render correctly", () => {
    const component = enzyme.shallow(
      <OrderBidChart data={getMockedData()} maxTotalSum={mockedMaxTotalSum} />,
    );

    expect(component).toMatchSnapshot();
  });
});

describe("OrderRow", () => {
  it("should render correctly", () => {
    const component = enzyme.shallow(
      <OrderRow
        backgroundColor="#123534"
        key="45000"
        priceText="45,000.0"
        priceTextColor="#129269"
        sizeText="100"
        totalPercentage={1}
        totalText="100"
      />,
    );

    expect(component).toMatchSnapshot();
  });
});

describe("SpreadRow", () => {
  it("should render correctly", () => {
    const component = enzyme.shallow(<SpreadRow units={100} percentage={10} />);

    expect(component).toMatchSnapshot();
  });
});

describe("Header", () => {
  it("should render correctly", () => {
    const component = enzyme.shallow(<Header />);

    expect(component).toMatchSnapshot();
  });
});

describe("ToggleProductButton", () => {
  it("should render correctly", () => {
    const onToggle = () => {};
    const component = enzyme.shallow(
      <ToggleProductButton onToggle={onToggle} />,
    );

    expect(component).toMatchSnapshot();
  });

  it("should call the callback function on press", () => {
    const mockCallback = jest.fn(() => {});
    const component = enzyme.shallow(
      <ToggleProductButton onToggle={mockCallback} />,
    );

    // Action
    component.find(Pressable).first().simulate("press");

    //Assert
    expect(mockCallback.mock.calls.length).toBe(1);
  });
});

describe("ErrorMessage", () => {
  it("should render correctly when display is true", () => {
    const component = enzyme.shallow(
      <ErrorMessage display={true} text={"Sample Error Message"} />,
    );

    expect(component).toMatchSnapshot();
  });

  it("should render correctly when display is false", () => {
    const component = enzyme.shallow(
      <ErrorMessage display={false} text={"Sample Error Message"} />,
    );

    expect(component).toMatchSnapshot();
  });
});
