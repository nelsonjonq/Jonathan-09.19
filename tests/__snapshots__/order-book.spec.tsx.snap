// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage should render correctly when display is false 1`] = `""`;

exports[`ErrorMessage should render correctly when display is true 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#3D1E28",
      "flexDirection": "row",
      "padding": 10,
      "paddingLeft": 15,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontWeight": "bold",
        "textAlign": "left",
      }
    }
  >
    Sample Error Message
  </Text>
</View>
`;

exports[`Header should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#3A404C",
        "borderBottomWidth": 1,
        "padding": 10,
        "paddingLeft": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontWeight": "bold",
        }
      }
    >
      Order Book
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "#3A404C",
          "borderBottomWidth": 1,
          "padding": 10,
          "paddingLeft": 15,
        },
        Object {
          "flexDirection": "row",
          "justifyContent": "space-evenly",
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#3A404C",
          "fontWeight": "bold",
        }
      }
    >
      PRICE
    </Text>
    <Text
      style={
        Object {
          "color": "#3A404C",
          "fontWeight": "bold",
        }
      }
    >
      SIZE
    </Text>
    <Text
      style={
        Object {
          "color": "#3A404C",
          "fontWeight": "bold",
        }
      }
    >
      TOTAL
    </Text>
  </View>
</View>
`;

exports[`OrderAskChart should render correctly 1`] = `
<View>
  <OrderRow
    backgroundColor="#3D1E28"
    key="45000"
    priceText="45,000.0"
    priceTextColor="#CB292E"
    sizeText="100"
    totalPercentage={9.5}
    totalText="950"
  />
  <OrderRow
    backgroundColor="#3D1E28"
    key="44950"
    priceText="44,950.0"
    priceTextColor="#CB292E"
    sizeText="200"
    totalPercentage={8.5}
    totalText="850"
  />
  <OrderRow
    backgroundColor="#3D1E28"
    key="44900"
    priceText="44,900.0"
    priceTextColor="#CB292E"
    sizeText="650"
    totalPercentage={6.5}
    totalText="650"
  />
</View>
`;

exports[`OrderBidChart should render correctly 1`] = `
<View>
  <OrderRow
    backgroundColor="#123534"
    key="45000"
    priceText="45,000.0"
    priceTextColor="#129269"
    sizeText="100"
    totalPercentage={1}
    totalText="100"
  />
  <OrderRow
    backgroundColor="#123534"
    key="44950"
    priceText="44,950.0"
    priceTextColor="#129269"
    sizeText="200"
    totalPercentage={3}
    totalText="300"
  />
  <OrderRow
    backgroundColor="#123534"
    key="44900"
    priceText="44,900.0"
    priceTextColor="#129269"
    sizeText="650"
    totalPercentage={9.5}
    totalText="950"
  />
</View>
`;

exports[`OrderRow should render correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "paddingBottom": 3,
      "paddingTop": 3,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "height": 24,
          "position": "absolute",
        },
        Object {
          "backgroundColor": "#123534",
          "width": "1%",
        },
      ]
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontWeight": "bold",
            "minWidth": 75,
            "textAlign": "right",
          },
          Object {
            "color": "#129269",
          },
        ]
      }
    >
      45,000.0
    </Text>
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontWeight": "bold",
          "minWidth": 75,
          "textAlign": "right",
        }
      }
    >
      100
    </Text>
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontWeight": "bold",
          "minWidth": 75,
          "textAlign": "right",
        }
      }
    >
      100
    </Text>
  </View>
</View>
`;

exports[`SpreadRow should render correctly 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#3A404C",
      "flexDirection": "row",
      "justifyContent": "center",
      "padding": 10,
      "paddingLeft": 15,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#3A404C",
        "fontWeight": "bold",
      }
    }
  >
    Spread 100.0 (10.00%)
  </Text>
</View>
`;

exports[`ToggleProductButton should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "margin": 25,
    }
  }
>
  <Pressable
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#5741D9",
        "borderRadius": 3,
        "height": 30,
        "justifyContent": "center",
        "width": 150,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Toggle Feed
    </Text>
  </Pressable>
</View>
`;
